datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -353,6 +354,20 @@ fn string_temporal_coercion(
     }
 }
 
+/// Coerce `Boolean` to other larger types, like Numeric as `1` or String as "1"
+fn bool_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataType> {
+    match (lhs_type, rhs_type) {
```

Comment:
I think this also needs to check when `rhs_type` is a `DataType::Boolean` as well. 

I would expect both of the following queries to work and return the same thing

```
sDataFusion CLI v33.0.0
❯ select 1 = true;
Error during planning: Cannot infer common argument type for comparison operation Int64 = Boolean
❯ select true = 1;
+--------------------------+
| Boolean(true) = Int64(1) |
+--------------------------+
| true                     |
+--------------------------+
1 row in set. Query took 0.006 seconds.
```